File: boardservice/src/main/java/com/workingbit/board/dao/BoardDao.java
package com.workingbit.board.dao;

import com.workingbit.board.config.AppProperties;
import com.workingbit.share.dao.BaseDao;
import com.workingbit.share.domain.impl.BoardContainer;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

/**
 * Created by Aleksey Popryaduhin on 18:16 09/08/2017.
 */
@Component
public class BoardDao extends BaseDao<BoardContainer> {

  @Autowired
  public BoardDao(AppProperties appProperties) {
    super(BoardContainer.class, appProperties.getRegion(), appProperties.getEndpoint(), appProperties.isTest());
  }
}

File: boardmodule/src/main/kotlin/com/workingbit/boardmodule/dao/BoardDao.kt
package com.workingbit.boardmodule.dao

import com.workingbit.boardmodule.config.BoardProperties
import com.workingbit.coremodule.dao.BaseDao
import com.workingbit.coremodule.domain.impl.Board
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import uy.kohesive.injekt.Injekt
import uy.kohesive.injekt.api.get

/**
 * Created by Aleksey Popryaduhin on 19:24 25/08/2017.
 */
@Component
class BoardDao @Autowired
constructor(appProperties: BoardProperties = Injekt.get())
    : BaseDao<Board>(Board::class.java, appProperties.region, appProperties.endpoint, appProperties.test)